### PR TITLE
Luca readme2

### PR DESCRIPTION
**Pull request by lucav76**

### CODE REVIEW

**Review by kkweon — APPROVED** [review]
Looks good to me :+1:

File: ros/src/tl_detector/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+The traffic light detection could be done in several ways, but we decided to use a neural network as it seemed a modern and flexible approach.
+
+Our pipeline is diveded in two stages:
+* Traffic Light detection, using a detector already trained on COCO
+* Traffic light classification, using a custom Neural Network
+
+Traffic Light boxes detection
+==
+For the Traffic Light Detection, we wanted to use a detector already trained. There are several in the Model Zoo of TensorFlow, able to hit seveal performance/precision points.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:12]: "to hit **several**..."

File: ros/src/tl_detector/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+![CNN Model](model.png)
+
+Attempts to use 4 layers seemd to show that the network was starting to overfit.
+
+## Training the Neural Network
+
+The Neural Network has been trained mainly using the Udacity dataset number 2. Unfortunately a relatively high amount of images were not classified correctly, so we had to manually clean the dataset.
+
+To make the Neural Network better are recognizing traffic lights in the simulator, We added some images taken from the simulator.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:38]: "better **at** recognizing..."

File: ros/src/tl_detector/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+During the tests, it become clear that the detector was recognizing also traffic lights oriented toward the other direction, so we added a class to classify that.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:40]: "it **became**..."

File: ros/src/tl_detector/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+* Every epoch the Augmenters will be asked to provide their image, so the cropping will be different every time, improving generalization
+
+DataSet and DataClasses have been put in the **LightNet** class, which known how the network needs to be trained, and just need to be given the directories with the images. The images are classified in directories, as it makes easier to find problems.
+
+The following diagram shows the main classes defined in the project:
+
+![Classes](classes.jpg)
+
+## Inference
+While testing, it become obvious that Keras models saved on Python 3 were not compatible with Python 2. To solve the problem, and hopefully increase performance, we froze the graph and exported it. In this way, while the training has been perfomed with Keras, the inference is done purely with TensorFlow.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:67]: "it **became**..."

File: ros/src/tl_detector/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+## Dealing with False Negatives
+The network has been designed to detect only one traffic light, as this was the requirement for Udacity. However it might detect mroe than one and skip the false positives.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:70]: "detect **more**..."

File: ros/src/tl_detector/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+A problem are however the false negative, as detector might have problems to find the traffic lights, sometimes.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:71]: "problem **is**..."

File: ros/src/tl_detector/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+A way to solve this issue, is to assume that if there are no detections, the last detection si still valid, at least for a few frames.
```
Comment (Krishtof-Korda) [inline ros/src/tl_detector/README.md:72]: "last detection **is**..."

**Pull request merged**